README.md

```diff
@@ -41,11 +41,18 @@ Mava allows researchers to experiment with multi-agent reinforcement learning (M
 
 ## Installation 🎬
 
-At the moment Mava is not meant to be installed as a library, but rather to be used as a research tool. We recommend cloning the Mava repo and pip installing as follows:
+At the moment Mava is not meant to be installed as a library, but rather to be used as a research tool. We recommend cloning the Mava repo and installing dependencies using [uv](https://github.com/astral-sh/uv) as follows:
 
 ```bash
-git clone https://github.com/instadeepai/mava.git
-cd mava
+# Clone the repository
+git clone https://github.com/instadeepai/Mava.git
+cd Mava
+# Create a virtual environment and install all dependencies
+uv sync
```

Comment:
I think we need to and an extra step here where the user either pins their global `uv` python version or pins the python version used in the Mava directory. This is for incase someone's base python version is 3.13 in which case the installation won't run. 

Another detail is that we need to add `setuptools` to the requirements and `uv.lock` or explicitly install it after running `uv sync` otherwise the code won't run. 

Might be overkill but adding a line activating the venv here could be nice as well.

Reply:
Good call on the setuptools stuff will do.
So uv will read our pyproject and take the latest python version that is compatible, we should actually limit our supported version to 3.12 and then sync will always create an env with the correct version

Reply:
I had a weird situation where I only had python 3.13 locally so then `uv sync` wouldn't work. But I don't think we have to explicitly mention this in the readme. Perhaps one day an error like that could live something like a FAQ.

Reply:
What we could do for the default install instructions in make it `uv sync -p 3.12`?